org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/antlr/internal/Lexer.java

```diff
@@ -25,17 +29,53 @@ protected Lexer(CharStream input) {
 	protected Lexer() {
 		super();
 	}
-	
+
 	protected Lexer(CharStream input, RecognizerSharedState state) {
 		super(input, state);
 	}
-	
+
+	@Override
+	public Token nextToken() {
+		while (true) {
+			state.token = null;
+			state.channel = Token.DEFAULT_CHANNEL;
+			state.tokenStartCharIndex = input.index();
+			state.tokenStartCharPositionInLine = input.getCharPositionInLine();
+			state.tokenStartLine = input.getLine();
+			state.text = null;
+			if ( input.LA(1)==CharStream.EOF ) {
+				return Token.EOF_TOKEN;
+			}
+			try {
+				mTokens();
+				if ( state.token==null ) {
+					emit();
+				}
+				else if ( state.token==Token.SKIP_TOKEN ) {
+					continue;
+				}
+				return state.token;
+			}
+			/*
+			 * Avoid infinite loop (editor freeze) on {@link FailedPredicateException}
+			 */
+			catch (NoViableAltException | FailedPredicateException e) {
+				reportError(e);
+				recover(e); // throw out current char and try again
+			}
+			catch (RecognitionException re) {
+				reportError(re);
+				// match() routine has already called recover()
```

Comment:
I do not understand the comment.

Reply:
The comment comes basically from the super implementation. The only modification I made on the super implementation is the extension of the catch block by  `| FailedPredicateException e`. So I would leave the comment as is to be able to get a readable diff between the super and the customized imlementation of the `nextToken()` method.